src/main/java/uk/gov/companieshouse/api/accounts/interceptor/TransactionInterceptor.java

```diff
@@ -28,26 +28,64 @@
     @Override
     @SuppressWarnings("unchecked")
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
-            Object handler) {
+        Object handler) {
         try {
             Map<String, String> pathVariables = (Map) request
-                    .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+                .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
             String transactionId = pathVariables.get("transactionId");
             ResponseEntity<Transaction> transaction = transactionManager
-                    .getTransaction(transactionId, request.getHeader("X-Request-Id"));
+                .getTransaction(transactionId, request.getHeader("X-Request-Id"));
             request.setAttribute(AttributeName.TRANSACTION.getValue(), transaction.getBody());
-            return isTransactionIsOpen(transaction);
+
+            return isValidTransaction(request, transaction);
+
         } catch (HttpClientErrorException httpClientErrorException) {
             response.setStatus(httpClientErrorException.getStatusCode().value());
             return false;
         }
     }
 
+    /**
+     * Check transaction is valid. e.g. Accounts api call needs the Transaction to be open. Filing
+     * Generator call needs the transaction to be closed.
+     *
+     * @param request
+     * @param transaction
+     * @return
+     */
+    private boolean isValidTransaction(HttpServletRequest request,
+        ResponseEntity<Transaction> transaction) {
```

Comment:
I would accept a `Transaction` and not the `ResponseEntity`. The method does nothing with the `ReponseEntity` object so you can just pass in the `Transaction` instead.